packages/tinybased/src/lib/types.ts

```diff
@@ -1,11 +1,44 @@
+import { SchemaBuilder } from './SchemaBuilder';
+import { CellSchema } from './TableBuilder';
+
 export type Cell = string | number | boolean;
 export type Table = Record<string, Cell>;
 export type TinyBaseSchema = Record<string, Table>;
+export type TableNames<TBSchema extends TinyBaseSchema> =
+  OnlyStringKeys<TBSchema>;
+
+export type InferSchema<SB> = SB extends SchemaBuilder<infer S, any>
+  ? S
+  : never;
 
 export type SchemaHydrators<TBSchema extends TinyBaseSchema> = {
   [TTableName in keyof TBSchema]: () => Promise<TBSchema[TTableName][]>;
 };
 
+export type SchemaHydrator<TBSchema extends TinyBaseSchema> = {
+  [TTableName in keyof TBSchema]: TTableName extends string
+    ? [TTableName, () => Promise<TBSchema[TTableName][]>]
+    : never;
+}[keyof TBSchema];
+
+export type PersisterSchema<TBSchema extends TinyBaseSchema> = {
```

Comment:
I feel like we could call this `TableDefs` or something like that instead to better draw a distinction